src/builder/QuarkBuilder.sol

```diff
@@ -574,8 +573,24 @@ contract QuarkBuilder {
         if (payment.isToken) {
             uint256 supplementalPaymentTokenBalance = 0;
             if (Strings.stringEqIgnoreCase(payment.currency, cometWithdrawIntent.assetSymbol)) {
-                // XXX in the withdrawMax case, use the Comet balance
-                supplementalPaymentTokenBalance += cometWithdrawIntent.amount;
+                if (isMaxWithdraw) {
+                    // when doing a maxWithdraw of the payment token, add the account's supplied balance
+                    // as supplemental payment token balance
+                    //
+                    // assumes that a maxWithdraw is for the base asset, since
```

Comment:
I did not include a check that the withdrawn token is the base asset. using an amount of `uint256.max` when withdrawing a collateral asset will revert.

this is also a check we can do in the as-yet-implemented check against your CometPositions, which we is unblocked now that we're passing in CometPosition info.

Reply:
Yeah it probably makes more sense to check that in the check against the user's CometPositions. So it's fair to assume `isMaxWithdraw` is for the base asset, as the earlier check should revert if not.